src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -104,8 +105,8 @@ struct sd_dhcp_client {
         usec_t expire_time;
         uint64_t attempt;
         uint64_t max_attempts;
-        OrderedHashmap *extra_options;
-        OrderedHashmap *vendor_options;
+        OrderedSet *extra_options;
```

Comment:
Nah, do not use `OrderedSet` for `extra_options`.

Reply:
why ?

Reply:
> why ?

As you can see in
> r = ordered_hashmap_ensure_put(&client->extra_options, &dhcp_option_hash_ops, UINT_TO_PTR(v->option), v);

the extra options must not have duplicated option number. I have not checked RFC, but at least, the current code refuse multiple options with the same option number.

Reply:
I have not seen a use case where same option can be duplicated. How the other end going to parse and which option code it will consider if it's duplicated. If multiple value is required then they are sent  like 
```
  Code   Len         Address 1               Address 2
   +-----+-----+-----+-----+-----+-----+-----+-----+--
   |  5  |  n  |  a1 |  a2 |  a3 |  a4 |  a1 |  a2 |  ...
   +-----+-----+-----+-----+-----+-----+-----+-----+--

```